src/main/java/io/axoniq/axonserver/connector/ResultStreamPublisher.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2022. AxonIQ
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.axoniq.axonserver.connector;
+
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Supplier;
+
+/**
+ * {@link Publisher} implementation that allows to interact with a {@link ResultStream}.
+ * <p>
+ * When subscribed, it generates a new {@link ResultStream} through the provided supplier.
+ * The generated {@link ResultStream} provides the elements to be published.
+ *
+ * @author Sara Pellegrini
+ * @since 4.6
+ */
+public class ResultStreamPublisher<Message> implements Publisher<Message> {
```

Comment:
The Generic Argument  "Message" is confusing, since that is also a commonly used class name in the connector. I'd recommend using something short, like "M" and document it in the Javadoc (that this is the type of element consumed from the underlying stream).

Reply:
Done.